ProefProeveVeluwe/Assets/Scripts/Runtime/GPS/GPS.cs

```diff
@@ -1,98 +1,65 @@
-        [SerializeField] private Text longitudeValue;
-        [SerializeField] private Text altitudeValue;
-        [SerializeField] private Text horizontalAccuracyValue;
-        [SerializeField] private Text timeStampValue;
-        [SerializeField] private Text updateRateValue;
-        
-        private int _currentTimer;
-        private int _maxTimer = 20;
-
-        private int _updateValue;
-
-        private void Start()
-        {
-            _currentTimer = _maxTimer;
-            StartCoroutine(GPSLocation());
-        }
-        
-        /// <summary>
-        /// When the player clicks on the retry button the coroutine will restart
-        /// </summary>
-        private void OnRetryClick()
-        {
-            _currentTimer = _maxTimer;
-            GPSStatus.text = "...";
-            StartCoroutine(GPSLocation());
-        }
+    public double CurrentLatitude;
+    public double CurrentLongitude;
+    public double CurrentAltitude;
+    public double HorizontalAccuracy;
+    public double TimeStamp;
 
-        /// <summary>
-        /// checks the device for location data
-        /// </summary>
-        private IEnumerator GPSLocation()
-        {
-            //check if location is enabled
-            if (!Input.location.isEnabledByUser) yield break;
-            Input.location.Start();
+    [SerializeField] private float _timer = 30f;
+    private float _currentTimer;
 
-            //wait until service initialize
-            while (Input.location.status == LocationServiceStatus.Initializing && _currentTimer > 0)
-            {
-                yield return new WaitForSeconds(1);
-                _currentTimer--;
-            }
+    public bool HasLocation = false;
 
-            // service didn't init in maxTimer sec
-            if (_currentTimer < 1)
-            {
-                GPSStatus.text = "Time Out";
-                yield break;
-            }
 
-            if (Input.location.status == LocationServiceStatus.Failed)
-            {
-                GPSStatus.text = "Unable to determine device location";
-                yield break;
-            }
+    public void Awake()
+    {
+        _currentTimer = _timer;
+        StartCoroutine(GPSLocation());
+    }
+    
+    private IEnumerator GPSLocation()
+    {
+        //check if location is enabled
+        if (!Input.location.isEnabledByUser) yield break;
+        Input.location.Start();
 
-            //Access granted
-            _updateValue++;
-            InvokeRepeating("UpdateGPSData", 0.5f, 1);
+        //wait until service initialize
+        while (Input.location.status == LocationServiceStatus.Initializing && _currentTimer > 0)
+        {
+            yield return new WaitForSeconds(1);
+            _currentTimer--;
         }
 
-        /// <summary>
-        /// Fills in text with the GPS Data 
-        /// </summary>
-        private void UpdateGPSData()
+        // service didn't init in maxTimer sec
+        if (_currentTimer < 1)
```

Comment:
What is the point of this if it cannot leave the while loop while _currentTimer is bigger than 0 in it cannot go here and after it is smaller that 0 it will go over here but then you check if it is smaller that 1 which by default it should already be?